homeassistant/components/vicare/entity.py

```diff
@@ -26,8 +27,10 @@ def __init__(
     ) -> None:
         """Initialize the entity."""
         gateway_serial = device_config.getConfig().serial
-        device_serial = device.getSerial()
-        identifier = f"{gateway_serial}_{device_serial}"
+        device_id = device_config.getId()
+        device_serial: str | None = get_device_serial(device)
```

Comment:
We don't want side effects in init methods.

Reply:
Would this be fine? 

https://github.com/home-assistant/core/pull/125495/commits/3845d1e5a0b7b6de25b538acc8fbe3414c3aafa9

Reply:
Yes, that's ok.